### PR TITLE
wgengine/tsdns: handle bonjour service prefixes in rdns names

### PR DESCRIPTION
**Pull request by josharian**
Noticed while debugging something else.

---
This change is [Reviewable](https://reviewable.io/reviews/tailscale/tailscale/782)

### CODE REVIEW

**Review by bradfitz — APPROVED** [review]

**Comment by danderson** [comment]
I don't follow why this is desirable. This would make us respond to Bonjour requests, even though we don't know what to respond with. Am I interpreting this right? Why do we want to respond with anything other than NXDOMAIN to these queries?

**Comment by josharian** [comment]
I could easily be missing a lot of context here...but if we know the answer, why wouldn't we respond to a Bonjour request? Any reason Bonjour shouldn't work over Tailscale?

**Comment by danderson** [comment]
We don't know the correct answer, do we? Those names are for a machine to query policy settings provided by the network administrator. What do we respond to them with this change? If I'm reading right, we chop off the Bonjour parts and respond with a record corresponding to some other name. Why?

**Comment by josharian** [comment]
Ah, I see now. I'll change this to just remove the logging noise, then.

**Comment by josharian** [comment]
Also added in a quick bug fix. PTAL.

File: wgengine/tsdns/forwarder.go [hunk 1]
```diff
@@ -143,8 +143,9 @@ func (f *forwarder) Start() error {
 	}
 
 	f.wg.Add(connCount + 1)
-	for idx, conn := range f.conns {
-		go f.recv(uint16(idx), conn)
+	for idx := range f.conns {
+		idx := idx
```
Comment (bradfitz) [inline wgengine/tsdns/forwarder.go:147]: this whole change is unnecessary.

arguments to goroutines are evaluated before the goroutine starts.

Comment (josharian) [reply wgengine/tsdns/forwarder.go]: Sigh. Yes, thanks. I just misread.

**Comment by josharian** [comment]
Submitting. Holler if you desire changes post-facto.

**Pull request merged**